modules/replication/module.py

```diff
@@ -151,6 +159,9 @@ def run(self):
                                 if (byz.is_byzantine() and
```

Comment:
I think we should have one surrounding `if byz.is_byzantine()` and then sub-ifs for each behaviour. That way we only need to log that node is byzantine once and it looks a bit cleaner.